src/libraries/Microsoft.Extensions.Configuration/tests/ConfigurationTest.cs

```diff
@@ -889,5 +889,28 @@ public void NullSectionDoesNotExist()
             // Assert
             Assert.False(sectionExists);
         }
+
+        internal class NullReloadTokenConfigSource : IConfigurationSource, IConfigurationProvider
+        {
+            public IEnumerable<string> GetChildKeys(IEnumerable<string> earlierKeys, string parentPath) => throw new NotImplementedException();
+            public Primitives.IChangeToken GetReloadToken() => null;
```

Comment:
Just a note for review, this PR is making sure we don't throw NRE when GetReloadToken is null.

Because otherwise as explained by @davidfowl in https://github.com/dotnet/runtime/issues/36393#issuecomment-531486938, rather than returning null in NullReloadTokenConfigSource in ConfigurationTest

```
public Primitives.IChangeToken GetReloadToken() => null;
```

If NullReloadTokenConfigSource.GetReloadToken() returned [NullChangeToken.Singleton](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.fileproviders.nullchangetoken.singleton?view=aspnetcore-2.2#Microsoft_Extensions_FileProviders_NullChangeToken_Singleton), then there would be no problem to solve.

Reply:
Issue #35996 mentioned that the docs suggest returning null. https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.configuration.iconfigurationprovider.getreloadtoken?view=dotnet-plat-ext-5.0